ext/DynamicQuantitiesLinearAlgebraExt.jl

```diff
@@ -0,0 +1,9 @@
+module DynamicQuantitiesLinearAlgebraExt
+
+import LinearAlgebra: norm
+import DynamicQuantities: AbstractQuantity, ustrip, dimension, new_quantity
+
+norm(q::AbstractQuantity, p::Real=2) = new_quantity(typeof(q), norm(ustrip(q), p), dimension(q))
+norm(q::AbstractArray{<:AbstractQuantity}, p::Real=2) = new_quantity(eltype(q), norm(ustrip(q), p), dimension(q))
```

Comment:
I read somewhere that dispatches such as this one, where there is a generic interface (`norm(q::AbstractArray)`) and one extends it in a case where one only owns some type parameters, but not the fully type, can be a bit iffy. I tried to find the discussion (I think this was said by @vtjnash) but it might have disappeared in the slack void:/ (I would love to find that discussion, as if true it would certainly be somewhat unwritten knowledge...) But I think practically speaking it can cause invalidations by pushing e.g. a number of eligible methods over `max_methods`. 

For this reason, since we don't / aren't able to make this performance optimization anywhere else, I would maybe nudge towards removing this method to avoid any invalidation headaches, unless `norm` is a particular bottleneck.

Reply:
Removal sounds good to me.

Reply:
Actually I think `norm` has to stay for arrays. I just tried removing it but it immediately runs into an error due to the use of `float` in `LinearAlgebra/src/generic.jl`:

```julia
generic_normInf(x) = float(mapreduce(norm, max, x))
```

which is used at the beginning of `norm2`:

```julia
function generic_norm2(x)
    maxabs = normInf(x)
    ...
```

One alternative would be to redefine the behavior of `float(::AbstractQuantity)` to only apply `float` to the value of a quantity. Right now `float` attempts to convert a quantity into a scalar, and throws an error if there are any dimensions left. Not sure what the best strategy is.

Reply:
Fixed in https://github.com/SymbolicML/DynamicQuantities.jl/pull/37